libs/ng-morph/ng/helpers/push-to-array-property.ts

```diff
@@ -27,11 +27,23 @@ export function pushToArrayProperty(
     return;
   }
 
+  if (!Node.isArrayLiteralExpression(property.getInitializer())) {
+    property.setInitializer(`[${property.getInitializer().getText()}]`);
+  }
+
```

Comment:
This is for cases when the constant is used, for example https://github.com/TinkoffCreditSystems/taiga-ui/blob/main/projects/demo/src/modules/app/app.browser.module.ts#L47
how do you like that? @IKatsuba 
Instead of just returning, wrap const in an array
providers: PROVIDERS -> providers: [PROVIDERS]

Reply:
боюсь что тут нужно подумать

Давай заведем ишью про принудительный перевод к массиву и отдельно на дубликаты? Сейчас фичи выглядят очень спорно

### Про массивы

По хорошему нужно понимать что там указанно вместо масссива и если это, например, ссылка на массив нужно проапдейтить оригинальный. Если это не массив, то лучше скипать либо выдавать варнинг, что "такой вот свойство непонятно что, но не массив. Используйте forceToArray" и добавть опцию forceToArray. 

### Про дубликаты

Пока сложно объяснить, но все таки я пока что против дополнительно проверки на дубликаты. И непонятно, нужно ли скипнуть совсем действие или перенести будликат в конец масссива. Короч есть сомнения. Возможно стоит добавить флаг про проверку на дубликаты

Reply:
слишком много сомнений для этого пиара. Можно поробовать похоливарить в отдельных ишью, либо сразу в пиарах. Предлагаю не тормохить этими спорами текущий

Reply:
Про массивы я исходил из логики, что на месте массива может быть только либо массив либо ссылка на массив, и можно объединить в массив из оригинального массива + новый элемент. Но согласен, можно ещё подумать, твой вариант основательнее, заведу отдельные ишью!

А по поводу дубликатов, из новой upsertImports тоже эту проверку убрать пока? а то выйдет, что, например addimportToNgModule продублирует модуль в imports, но не продублирует сам импорт, если он есть уже. Или там оставить?

Reply:
> А по поводу дубликатов, из новой upsertImports тоже эту проверку убрать пока? а то выйдет, что, например addimportToNgModule продублирует модуль в imports, но не продублирует сам импорт, если он есть уже. Или там оставить?

upsertImports - если уже есть импорт, то апдейтим, если нет инсертим новый

Reply:
добавил add-provider-to-component, add-style-url-to-component, add-provider-to-directive, предлагаю работу с другими дескрипторами тоже в другой пр перенести, там тоже надо будет подумать